htlcswitch/switch_test.go

```diff
@@ -1726,20 +1726,58 @@ func TestSwitchSendPayment(t *testing.T) {
 	// Handle the request and checks that bob channel link received it.
 	errChan := make(chan error)
 	go func() {
-		_, err := s.SendHTLC(
+		err := s.SendHTLC(
 			aliceChannelLink.ShortChanID(), 0, update,
 			newMockDeobfuscator())
-		errChan <- err
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		result, err := s.GetPaymentResult(0)
```

Comment:
Alternatively, what about returning a future here? Otherwise it would seem that the router needs to continually poll this method.